### PR TITLE
Add org.kde.ark (Ark)

### PR DESCRIPTION
**Pull request by Siosm**

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
bot, build org.kde.ark

**Comment by Siosm** [comment]
Mention @tsdgeos as he currently is a maintainer of several KDE related packages on Flathub.

**Comment by Siosm** [comment]
bot, build org.kde.ark

**Comment by Siosm** [comment]
bot, build org.kde.ark

**Comment by tsdgeos** [comment]
Looks good to me :)

**Comment by Siosm** [comment]
LGTM too!

File: org.kde.ark.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "command": "ark",
+    "rename-icon": "ark",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
+        "--device=dri"
+    ],
+    "modules": [
```
Comment (barthalion) [inline org.kde.ark.json:15]: You should cleanup unneeded files, at least `/include`, `/share/man`, `*.a` and `*.la` files at first glance.

**Comment by Erick555** [comment]
@Siosm p7zip plugin is missing. Also lack of `org.a11y.Bus` permission.

File: org.kde.ark.json (hunk 1 above)
Comment (Erick555) [reply org.kde.ark.json:15]: Also `pkgconfig`

File: org.kde.ark.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    "runtime": "org.kde.Platform",
+    "runtime-version": "5.14",
+    "sdk": "org.kde.Sdk",
+    "command": "ark",
+    "rename-icon": "ark",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
Comment (Erick555) [inline org.kde.ark.json:12]: Is filesystem=host really needed while this app should work just fine with portals?

Comment (tsdgeos) [reply org.kde.ark.json:12]: AFAIK the portal is just "give me one file to work with", no?

That's fine for opening an archive, but how is that going to unarchive something?

Comment (Erick555) [reply org.kde.ark.json:12]: You are right, portals still suck.

Comment (Siosm) [reply org.kde.ark.json:12]: This could potentially be removed but would require some changes in the App. I don't think this is a priority right now.

**Comment by Siosm** [comment]
bot, build org.kde.ark

**Comment by Siosm** [comment]
bot, build org.kde.ark

File: org.kde.ark.json (hunk 1 above)
Comment (Siosm) [reply org.kde.ark.json]: Updated

**Comment by tsdgeos** [comment]
I'm really disappointed that you submtted a yaml without discussing it as i asked, you're basically making my job harder just because you feel like.

Pissing off people is not going to help your "let's bring all kde appliations to flathub" goal.

@barthalion please do not merge any more org.kde.* stuff untill we have this discussion.

**Comment by barthalion** [comment]
@tsdgeos Sorry, I haven't followed the discussion close enough to notice there are unresolved problems. I'll wait for your explicit approval for the rest. (FYI, you have write access to the Flathub repo for Ark too, if you want to switch it back to JSON.)

**Comment by Siosm** [comment]
First, I would like to say that I am really sorry about how this discussion turned out. It was not my intention to make it look like I was ignoring your concerns. I should have more clearly stated that those PRs were still work in progress and that they should not have been merged.

That been said, I don't find you response appropriate given that:
  * I did not ask for the PRs to get merged.
  * I did not request nor implicitly expected that you took part in maintaining the apps I submitted.
  * I pushed YAML versions to test them in CI as local testing of manifests appeared to be not sufficient in the past.
  * I asked about your concerns regarding JSON/YAML and I offered two times to work with you on moving forward.
  * This is my first contribution to the Flathub community and I do not really feel welcomed.
  * I made a public call for action on the Flathub Discourse and I welcome any guidance from KDE developers.
  * I acted on review feedback and I am making sure that the necessary changes are contributed back to upstream repositories.
  * Just like probably most of you I am doing this during my free time.

I would really appreciate if we could have a constructive discussion around KDE Apps maintenance on Flathub and I will thus create a topic on the Flathub Discourse to gather people interested in maintaining KDE applications on Flathub.

**Comment by tsdgeos** [comment]
> I welcome any guidance from KDE developers.
Do you? Because to me it feels my guidance is being kind of ignored

> I did not request nor implicitly expected that you took part in maintaining the apps I submitted.
That's kind of weird too, you keep saying you want to "form a team" and at the same team you're asking people from the existing team (me) to not help?

> Just like probably most of you I am doing this during my free time.
That's never an excuse to do things wrong.

**Comment by Erick555** [comment]
@Siosm I'm generally grateful for your work here and elsewhere. I think the problem is that all reviews happened on json version (even @tsdgeos [gave you approval](https://github.com/flathub/flathub/pull/1577#issuecomment-643676120)) then you switched to yaml which was [hard nack](https://github.com/flathub/flathub/pull/1579#issuecomment-644382921) in other thread. In result it misled flathub maintainers who merged this prematurely.

> I did not ask for the PRs to get merged.

Well, you opened PR and said it's [LGTM](https://github.com/flathub/flathub/pull/1577#issuecomment-643807010). It's hard to interpretate this in other way than ask to merge even if you silently changed your mind after that.

> I asked about your concerns regarding JSON/YAML and I offered two times to work with you on moving forward.

I believe the best way forward is to go back to json for this and other apps. You may keep discussion about yaml conversion open in other channels but putting all your precious work in limbo because of this would be a waste.

**Pull request closed without merging**